### ISSUE TITLE
Apache 2.4 fails to start on CentOS 7

### ISSUE THREAD

**Issue by xyntrix** [echocat/puppet-graphite#212]
With a fairly basic configuration, Apache fails to start with "address already in use" due to the graphite_ports.conf symlinking back to the main apache ports config file.  To fix, just don't create this symlink.  Currently workaround, we just remove the symlink.

lrwxrwxrwx 1 root   root     21 Oct 12 15:14 graphite_ports.conf -> /etc/httpd/ports.conf
````
-- Logs begin at Fri 2015-10-09 14:04:36 MST, end at Mon 2015-10-12 15:42:06 MST. --
Oct 12 15:42:06 somehost.example.tld httpd[26445]: [Mon Oct 12 15:42:06.317119 2015] [so:warn] [pid 26445] AH01574: module wsgi_module is already loaded, skipping
Oct 12 15:42:06 somehost.example.tld httpd[26445]: AH00548: NameVirtualHost has no effect and will be removed in the next release /etc/httpd/conf.d/graphite_ports.conf:8
Oct 12 15:42:06 somehost.example.tld httpd[26445]: (98)Address already in use: AH00072: make_sock: could not bind to address 0.0.0.0:80
Oct 12 15:42:06 somehost.example.tld httpd[26445]: no listening sockets available, shutting down 
Oct 12 15:42:06 somehost.example.tld httpd[26445]: AH00015: Unable to open logs 
Oct 12 15:42:06 somehost.example.tld systemd[1]: httpd.service: main process exited, code=exited, status=1/FAILURE
Oct 12 15:42:06 somehost.example.tld kill[26447]: kill: cannot find process ""
Oct 12 15:42:06 somehost.example.tld systemd[1]: httpd.service: control process exited, code=exited status=1
Oct 12 15:42:06 somehost.example.tld systemd[1]: Failed to start The Apache HTTP Server.
-- Subject: Unit httpd.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit httpd.service has failed.
-- 
-- The result is failed.
````

snip from basic config:

````
     gr_user => graphite,
     gr_group => graphite,
      gr_web_server => 'apache',
      gr_web_servername => 'foohost.example.com',
      gr_web_user => apache,
      gr_web_group => apache,
      gr_apache_port => 80,
      gr_apache_port_https => 443,
      gr_line_receiver_interface => '127.0.0.1',
      gr_relay_pickle_interface => '127.0.0.1',
      gr_aggregator_line_interface => '127.0.0.1',
      gr_cache_query_interface => '127.0.0.1',
      gr_relay_pickle_port => 2203,
      gr_line_receiver_port => 2213,
      gr_aggregator_line_port => 2223,
      gr_cache_query_port => 7203,
````

**Comment by xyntrix** [comment]
Ahhh.  Looks like the file resource ignores gr_apache_port if it's an int vs a string:
```
      if $::graphite::gr_apache_port != '80' {
        file { "${::graphite::params::apacheconf_dir}/${::graphite::params::apacheports_file}":
````

I changed my manifest to:      gr_apache_port => '80' instead of     gr_apache_port => 80,    
and now it's not creating the symlink file.   It might be good to test for both of these cases.

**Comment by dwerder** [comment]
We should check if this is a integer. So a port should be an integer at all ;-)

**Comment by brettswift** [comment]
#217  doesn't fix this for me on CentOS 6.x.  
If we're validating an integer, that conditional should also check for an integer.  When I run this on a clean server the ports still collide.

This might be the biggest pull request in history - but if you'd like it - I've removed the quotes. :) #219

**Comment by dwerder** [comment]
Closed with PR #219 in version 5.15.0
